tests/test_json_validation.py

```diff
@@ -0,0 +1,75 @@
+from jsonschema.validators import _utils, extend
+
+import pytest
+from conftest import build_app_from_fixture
+from connexion import App
+from connexion.decorators.validation import RequestBodyValidator
+from connexion.json_schema import Draft4RequestValidator
+
+SPECS = ["swagger.yaml", "openapi.yaml"]
+
+@pytest.mark.parametrize("spec", SPECS)
+def test_validator_map(json_validation_spec_dir, spec):
+    def validate_type(validator, types, instance, schema):
+        types = _utils.ensure_list(types)
+        errors = Draft4RequestValidator.VALIDATORS['type'](validator, types, instance, schema)
+        for e in errors:
+            yield e
+
+        if 'string' in types and 'minLength' not in schema:
+            errors = Draft4RequestValidator.VALIDATORS['minLength'](validator, 1, instance, schema)
+            for e in errors:
+                yield e
+
+    MinLengthRequestValidator = extend(Draft4RequestValidator, {'type': validate_type})
+
+    class MyRequestBodyValidator(RequestBodyValidator):
+        def __init__(self, *args, **kwargs):
+            super(MyRequestBodyValidator, self).__init__(*args, validator=MinLengthRequestValidator, **kwargs)
+
+    validator_map = {'body': MyRequestBodyValidator}
+
+    app = App(__name__, specification_dir=json_validation_spec_dir)
+    app.add_api(spec, validate_responses=True, validator_map=validator_map)
+    app_client = app.app.test_client()
+
+    res = app_client.post('/v1.0/minlength', json={'foo': 'bar'}) # type: flask.Response
```

Comment:
I think you might have to use `data=json.dumps({"foo": "bar"})` and `headers={"Content-Type": "application/json"}`

Reply:
These lines are causing CI to fail: https://travis-ci.org/zalando/connexion/jobs/430130461#L984
I said "might", because I haven't tested it (or maybe there's a better fix).

Reply:
Hmm, tox installed flask=0.10.1, on my maschine tox gave me 1.0.2